docs: Add instructions for installing dependencies and change build directory to avoid mixing Spider & dependencies. (#122)
y-scope/spider: docs/src/user-docs/guides-quick-start.md

you'll need to:

* Write a task

> * (Optional) Install Dependencies

Reviewer: ```suggestion
* Install Spider's dependencies
```

Installing Spider's dependencies isn't actually optional. Using the task is the optional part.

* [Docker] 20.10+
  * If you're not running as root, ensure `docker` can be run
    [without superuser privileges][docker-non-root].

> * [Task](https://taskfile.dev/) v3.30.0+ if you want to install dependencies locally.

Reviewer: ```suggestion
* [Task] v3.30.0+ if you want to install Spider's dependencies within its build directory.
```

Let's move the link to the references section at the end of the doc.

The `SPIDER_REGISTER_TASK` macro at the bottom of `src/tasks.cpp` is how we inform Spider that a
function should be treated as a task.

> # (Optional) Installing dependencies

Reviewer: ```suggestion
# Installing dependencies
```

> You can install `Spider` dependencies locally using the `task` command by running:

Reviewer: ```suggestion
You can install `Spider` dependencies within its build directory by running:
```

```shell

> task deps:lib_install

Reviewer: The command in the README is wrong. Let's fix it in this PR as well.

```

This will install all dependencies in the `build/deps` directory.

> Alternatively, you can install the dependencies yourself. Check the `CMakeLists.txt` file for the
> list of dependencies.

Reviewer: ```suggestion
Alternatively, you can install the dependencies to the system. See the `install-all-run` task in
[deps-task.yaml] for the list of dependencies to install.
```

Add a link to the file on GitHub (we can't link to it locally since Sphinx will turn it into a download link).

To build the scheduler, run:

```shell

> cmake --build build/spider --parallel $(nproc) --target spider_scheduler

Reviewer: There are some instance that were missed. E.g., the one below should be `build/spider/spider`. I'd recommend going through the quick-start guide and making sure every command is correct.
